.github/CONTRIBUTING.md

```diff
@@ -25,22 +25,54 @@ You'll see exactly what is sent, be able to add your comments, and then send it.
-* Please show intention to maintain your features and code after you contribute a PR. Unmaintained code is a hassle for core developers. If you do not intend to maintain features you plan to contribute, please rethink your submission, or clearly state that in the PR description.
+* In particular **do not bring non-free software** (e.g. binary blobs) into the project. Make sure you do not introduce Google closed-source libraries.
+
+### Before starting development
+
+* If you want to help out with an existing bug report or feature request, **leave a comment** on that issue saying you want to try your hand at it.
+* If there is no existing issue for what you want to work on, **open a new one** describing your changes. This gives the team and the community a chance to give **feedback** before you spend time on something that is already in development, should be done differently, or should be avoided completely.
+* Please show **intention to maintain your features** and code after you contribute a PR. Unmaintained code is a hassle for core developers. If you do not intend to maintain features you plan to contribute, please rethink your submission, or clearly state that in the PR description.
+* Send PRs that **only cover one specific issue/solution/bug**. Do not send PRs that are huge and could be split into multiple independent contributions.
+
+### Creating a Pull Request (PR)
+
+* Make changes on a **separate branch** with a meaningful name, not on the _master_ branch or the _dev_ branch. This is commonly known as *feature branch workflow*. You may then send your changes as a pull request (PR) on GitHub.
+* Please **test** (compile and run) your code before submitting changes! Ideally, provide test feedback in the PR description. Untested code will **not** be merged!
 * Respond if someone requests changes or otherwise raises issues about your PRs.
-* Send PRs that only cover one specific issue/solution/bug. Do not send PRs that are huge and consist of multiple independent solutions.
 * Try to figure out yourself why builds on our CI fail.
+* Make sure your PR is **up-to-date** with the rest of the code. Often, a simple click on "Update branch" will do the job, but if not, you must *rebase* the dev branch manually and resolve the problems on your own. You can find help [on the wiki](https://github.com/TeamNewPipe/NewPipe/wiki/How-to-merge-a-PR). That makes the maintainers' jobs way easier.
+
+## Build instructions
+
+### Basic setup
+
+NewPipe is developed using [Android Studio](https://developer.android.com/studio/). The setup is fairly simple:
+- Clone the NewPipe repository with `git clone https://github.com/TeamNewPipe/NewPipe.git` (or use your fork)
+- Open the folder you just cloned with Android Studio
+- Build and run just like you would do with any other app, with the green triangle in the top bar
```

Comment:
I wish it was that simple… You forgot to accept license.

Reply:
What? With any Android Studio project I clone I just open it and it works fine out of the box (unless the project uses some strange gradle tasks)

Reply:
"(or use your fork)"

Do you mean "Use the Fork button"?

Reply:
Not the first time, you have to accept the license

Reply:
Well, then we should link to a guide on how to setup Android Studio, I wouldn't talk about the whole AS config here but just how to import NewPipe. Any good link?